src/hooks/useGridKeyboardNavigation/__tests__/useGridKeyboardNavigation.jest.js

```diff
@@ -0,0 +1,100 @@
+import { renderHook, cleanup, act } from "@testing-library/react-hooks";
+import { fireEvent } from "@testing-library/react";
+import range from "lodash/range";
+import useGridKeyboardNavigation from "../useGridKeyboardNavigation";
+import { NAV_DIRECTIONS } from "../../useFullKeyboardListeners";
+
+describe("useGridKeyboardNavigation", () => {
+  let element;
+
+  afterEach(() => {
+    element.remove();
+    cleanup();
+  });
+
+  function itemsArray(length) {
+    return range(length);
+  }
+
+  function renderHookForTest({ items = itemsArray(4), numberOfItemsInLine = 3, onItemClicked = jest.fn(), focusOnMount = false }) {
+    const itemsCount = items.length;
+    const getItemByIndex = index => items[index];
+
+    element = document.createElement("div");
+    document.body.appendChild(element);
+
+    return renderHook(
+      () =>
+        useGridKeyboardNavigation({
+          ref: { current: element },
+          itemsCount,
+          getItemByIndex,
+          onItemClicked,
+          focusOnMount,
+          numberOfItemsInLine
+        })
+    );
+  }
+
+  it("should set the active index to 0 when focusing for the first time", () => {
+    const { result } = renderHookForTest({ });
+
+    act(() => element.dispatchEvent(new Event('focus')));
```

Comment:
can we use `"@testing-library/user-event"` for the events?.

Reply:
I don't think it's possible. the closest thing is `userEvent.tab`, but it's not useful here.
(the `CustomEvent` is also irrelevant for `userEvent`)

Reply:
Don't forget that if you need real focus and not only trigger you need `element.focus()`
"Dispatching the event will not set the focus, that will just trigger the attached event handlers."

Reply:
that's a good idea. I tried it though and the event handlers weren't triggered during the test. I suspect that this happens because of the testing environment (on the browser it works).
In any case, I only need to test the handlers, so I'm fine with it as is :)